crates/bevy_ecs/src/bundle.rs

```diff
@@ -466,7 +467,7 @@ impl BundleInfo {
                         // the target table contains the component.
                         unsafe { table.get_column_mut(component_id).debug_checked_unwrap() };
                     // SAFETY: bundle_component is a valid index for this bundle
-                    match bundle_component_status.get_status(bundle_component) {
+                    match unsafe { bundle_component_status.get_status(bundle_component) } {
```

Comment:
I really do not like having this in the match expression, could we lift this out to before the match expression?